data-analytics/iot-nirvana/app-engine/src/main/java/com/google/cloud/demo/iot/nirvana/shared/Utils.java

```diff
@@ -0,0 +1,373 @@
+      Map<String, String> parametersMap,
+      long delay) {
+
+    // prepare task options
+    final TaskOptions taskOptions =
+        TaskOptions.Builder.withUrl(taskUrl).taskName(taskName).method(method);
+
+    // add parameters
+    for (String key : parametersMap.keySet()) {
+      taskOptions.param(key, parametersMap.get(key));
+    }
+
+    // add eventual delay
+    if (delay > 0) {
+      taskOptions.countdownMillis(delay);
+    }
+
+    // create the queue
+    final Queue queue = QueueFactory.getQueue(queueName);
+
+    Callable<Boolean> callable =
+        new Callable<Boolean>() {
+          public Boolean call() throws Exception {
+            queue.add(taskOptions);
+            return true;
+          }
+        };
+
+    Retryer<Boolean> retryer =
+        RetryerBuilder.<Boolean>newBuilder()
+            .retryIfException()
+            .withWaitStrategy(WaitStrategies.exponentialWait(100, 5, TimeUnit.MINUTES))
+            .withStopStrategy(StopStrategies.stopAfterAttempt(5))
+            .build();
+    try {
+      retryer.call(callable);
+    } catch (RetryException e) {
+      LOG.warning("enqueueTask() failed.\nStackTrace: " + Throwables.getStackTraceAsString(e));
+    } catch (ExecutionException e) {
+      LOG.warning("enqueueTask() failed.\nStackTrace: " + Throwables.getStackTraceAsString(e));
+    }
+
+    return;
+  }
+
+  /**
+   * Get headers of a HttpServletRequest
+   *
+   * @param request the request to be parsed
+   * @return map of request headers
+   */
+  public static Map<String, String> getHeadersInfo(HttpServletRequest request) {
+    Map<String, String> map = new HashMap<String, String>();
+    Enumeration headerNames = request.getHeaderNames();
+    while (headerNames.hasMoreElements()) {
+      String key = (String) headerNames.nextElement();
+      String value = request.getHeader(key);
+      map.put(key, value);
+    }
+    return map;
+  }
+
+  /**
+   * Get value of a specific cookie in the request
+   *
+   * @param requestthe request that contains the cookie
+   * @param cookieName the name of the cookie that you want to get
+   * @return the value of the cookie in the request
+   */
+  public static String getCookieValue(HttpServletRequest request, String cookieName) {
+    Cookie[] cookies = request.getCookies();
+    if (cookies != null) {
+      for (Cookie cookie : cookies) {
+        if (cookie.getName().equals(cookieName)) {
+          return cookie.getValue();
+        }
+      }
+    }
+    // no cookie found
+    return null;
+  }
+
+  /**
+   * Get JSON payload
+   *
+   * @param request the request that has to be processed
+   * @return payload of the request
+   * @throws IOException
+   */
+  public static JsonObject getJsonPayload(HttpServletRequest request) throws IOException {
+    JsonParser jsonParser = new JsonParser();
+    String body = null;
+    StringBuilder stringBuilder = new StringBuilder();
+    BufferedReader bufferedReader = null;
+
+    try {
```

Comment:
Shorter version of read is better.

Reply:
Removed the entire getJsonPayload method, it was not used. Change in commit https://github.com/GoogleCloudPlatform/professional-services/pull/61/commits/bdfede8409eb6660d48b75bcdb1897bad3757c02.